letta update fails on Windows — spawn npm ENOENT (needs shell: true for 
pm.cmd)

## Summary

`letta update` fails on Windows with `spawn npm ENOENT`. The root cause is in `src/updater/auto-update.ts`: it calls `execFileAsync("npm", ...)` without `shell: true`. On Windows, `npm` is `npm.cmd` — a batch file, not an executable. `execFile` can't resolve `.cmd` files without a shell, so it throws ENOENT before npm even starts.

This affects every Windows install on every supported Node version (`node >= 22.19.0`) — not specific to any particular setup.

## Root Cause

Four callsites in `src/updater/auto-update.ts` call `execFileAsync` without `shell: true`:

```typescript
// getNpmGlobalPath() — line ~155
const { stdout } = await execFileAsync("npm", ["prefix", "-g"], { timeout: 5000 });

// performUpdate() — line ~185 (primary), ~196 (ENOTEMPTY retry), ~210 (race-condition retry)
await execFileAsync(pm, installArgs, { timeout: 60000 });
```

All supported Node versions (`node >= 22.19.0`) block `.cmd` file execution via `execFile` without `shell: true`, throwing `EINVAL`.

## Reproduction

**Environment:**
- OS: Windows 10 (Build 10.0.26200), x64
- Node.js: v25.2.1 (installed via nvm-windows; same behavior with the official Node.js installer — both ship `npm.cmd`, not `npm.exe`)
- npm: 11.16.0
- Letta Code: 0.27.2

**Steps:**

1. `npm install -g @letta-ai/letta-code`
2. `letta update`

**Expected:** Update succeeds or reports "already on latest version."

**Actual:** `Update failed: spawn npm ENOENT`

**Minimal reproduction** (isolates the exact upstream code path):

```javascript
const { execFile } = require("node:child_process");
const { promisify } = require("node:util");
const execFileAsync = promisify(execFile);

(async () => {
  // Reproduces getNpmGlobalPath()
  try {
    const { stdout } = await execFileAsync("npm", ["prefix", "-g"], { timeout: 5000 });
    console.log("getNpmGlobalPath:", stdout.trim());
  } catch (e) {
    console.log("getNpmGlobalPath: FAIL —", e.code);
  }

  // Reproduces performUpdate() spawn (using --version instead of install for safety)
  try {
    const { stdout } = await execFileAsync("npm", ["--version"], { timeout: 5000 });
    console.log("performUpdate spawn:", stdout.trim());
  } catch (e) {
    console.log("performUpdate spawn: FAIL —", e.code);
  }
})();
```

**Output on Windows:** `ENOENT` for both calls.

## Suggested Fix

Add `shell: process.platform === "win32"` to all four `execFileAsync` calls:

```typescript
// getNpmGlobalPath()
const { stdout } = await execFileAsync("npm", ["prefix", "-g"], {
  timeout: 5000,
  shell: process.platform === "win32",
});

// performUpdate() — remaining three callsites (primary, ENOTEMPTY retry, race-condition retry)
await execFileAsync(pm, installArgs, {
  timeout: 60000,
  shell: process.platform === "win32",
});
```

## Compounding Issues

Fixing the ENOENT lets the update start, but peer dependency conflicts may still affect the result on Windows — `ink@7.0.5` requires `react@>=19.2.0`, but the package ships `react@18.2.0`. npm overrides these (exit 0), but the ERESOLVE warnings indicate a dependency tree mismatch. (See #1998.)

## Related

- #1998 — Peer-dep conflict (`ink-text-input@5.0.1` peer `ink@^4` vs root `ink@^5`) and a related Windows spawn bug in `installChannelRuntime` (`spawn EINVAL`). Same root cause for the spawn issue, different code path. Would be great to address both spawn bugs together.

--- a/src/updater/auto-update.test.ts
+++ b/src/updater/auto-update.test.ts
@@ -5,6 +5,7 @@ import * as path from "node:path";
 import {
   buildInstallCommand,
   buildLatestVersionUrl,
+  buildUpdateExecOptions,
   checkForUpdate,
   detectPackageManager,
   getSelfUpdateStatus,
@@ -271,6 +272,22 @@ describe("getSelfUpdateStatus", () => {
 });
 
 describe("update config resolution", () => {
+  test("buildUpdateExecOptions enables shell on Windows for npm.cmd shims", () => {
+    expect(buildUpdateExecOptions(5000, "win32")).toEqual({
+      timeout: 5000,
+      encoding: "utf8",
+      shell: true,
+    });
+  });
+
+  test("buildUpdateExecOptions keeps direct execFile behavior off Windows", () => {
+    expect(buildUpdateExecOptions(60000, "darwin")).toEqual({
+      timeout: 60000,
+      encoding: "utf8",
+      shell: false,
+    });
+  });
+
   test("resolveUpdatePackageName uses default when unset", () => {
     expect(resolveUpdatePackageName({} as NodeJS.ProcessEnv)).toBe(
       "@letta-ai/letta-code",

--- a/src/updater/auto-update.ts
+++ b/src/updater/auto-update.ts
@@ -1,4 +1,7 @@
-import { execFile } from "node:child_process";
+import {
+  type ExecFileOptionsWithStringEncoding,
+  execFile,
+} from "node:child_process";
 import { accessSync, constants, realpathSync } from "node:fs";
 import { readdir, rm } from "node:fs/promises";
 import { dirname, join } from "node:path";
@@ -114,6 +117,17 @@ export function buildInstallCommand(
   env: NodeJS.ProcessEnv = process.env,
 ): string {
   return `${pm} ${buildInstallArgs(pm, env).join(" ")}`;
+}
+
+export function buildUpdateExecOptions(
+  timeout: number,
+  platform: NodeJS.Platform = process.platform,
+): ExecFileOptionsWithStringEncoding {
+  return {
+    timeout,
+    encoding: "utf8",
+    shell: platform === "win32",
+  };
 }
 
 function getResolvedEntrypoint(): string {
@@ -316,9 +330,11 @@ export async function checkForUpdate(
  */
 async function getNpmGlobalPath(): Promise<string | null> {
   try {
-    const { stdout } = await execFileAsync("npm", ["prefix", "-g"], {
-      timeout: 5000,
-    });
+    const { stdout } = await execFileAsync(
+      "npm",
+      ["prefix", "-g"],
+      buildUpdateExecOptions(5000),
+    );
     return stdout.trim();
   } catch {
     return null;
@@ -370,7 +386,7 @@ async function performUpdate(progressLog?: (message: string) => void): Promise<{
   try {
     debugLog(`Running ${installCmd}...`);
     progressLog?.(`Running update command: ${installCmd}`);
-    await execFileAsync(pm, installArgs, { timeout: 60000 });
+    await execFileAsync(pm, installArgs, buildUpdateExecOptions(60000));
     debugLog("Update completed successfully");
     progressLog?.("Update command completed successfully.");
     return { success: true };
@@ -390,7 +406,7 @@ async function performUpdate(progressLog?: (message: string) => void): Promise<{
       await cleanupOrphanedDirs(globalPath);
 
       try {
-        await execFileAsync(pm, installArgs, { timeout: 60000 });
+        await execFileAsync(pm, installArgs, buildUpdateExecOptions(60000));
         debugLog("Update succeeded after cleanup retry");
         return { success: true };
       } catch (retryError) {
@@ -429,7 +445,7 @@ async function performUpdate(progressLog?: (message: string) => void): Promise<{
         await cleanupOrphanedDirs(globalPath);
       }
       try {
-        await execFileAsync(pm, installArgs, { timeout: 60000 });
+        await execFileAsync(pm, installArgs, buildUpdateExecOptions(60000));
         debugLog("Update succeeded after race condition retry");
         return { success: true };
       } catch (retryError) {

